ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,266 @@
+# 8. HACBS Test Stream - API contracts
+
+Date: 2022-11-28
+
+## Status
+
+Accepted
+
+## Context
+
+The Hybrid Application Cloud Build Service being developed aims to serve Red Hat teams but also partners and customers. This requires a level of adaptability to avoid recreating custom flows and Tasks for each stakeholder.
+
+In this respect Tasks developed by HACBS test stream should allow swapping external systems to accommodate different environments. This swap should not induce the complete recreation of pipelines.
+
+This and the idea of providing a homogeneous experience, which is easier to comprehend and navigate complex systems, leads to the definition of `API contracts`. These contracts need to be understood as guidance that may evolve with time and experience while keeping the aim of building a flexible homogeneous system.
+
+Tasks from the test stream may exchange information through
+
+1. Parameters passed at the time of the trigger of a PipelineRun (that is user, git repository, etc)
+2. Output of upstream elements of the Pipeline which may be needed for performing the Task (that is build artifacts, container image, etc)
+3. Providing whether the Task was successful or not
+4. Output of the Task that may be needed for processing subsequent Tasks of the pipeline
+5. Output of the Task that need to be stored for auditing and troubleshooting (that is summarized results of a validation or a scan, etc)
+6. Communication with external systems that are needed for performing the Task (that is container image repository, image scanner, etc)
+7. Connection details and credentials that may be required for the above
+
+## Decision
+
+### Results and Records
+
+Related to [PLNSRVCE-41](https://issues.redhat.com/browse/PLNSRVCE-41) investigations in regards to [Tekton results](https://github.com/tektoncd/results). The investigations are documented [here](https://docs.google.com/document/d/1WGWx6MllVs-BwRLd0PP5QTQHgGp0fW2dOYRKRAKWiwo/edit#heading=h.bpyxfhyud4dv).
```

Comment:
Move the information in this sentence to the `Context` section please. It provides background on why and how we came to the `Decision`.

Reply:
Done. Thanks.